docs/page-objects/testing.rst

```diff
@@ -0,0 +1,105 @@
+
+Each dependency is serialized to one or several ``bytes`` objects, each of
+which is saved as a single file. :func:`web_poet.serialization.serialize_leaf`
+and :func:`web_poet.serialization.deserialize_leaf` are used to convert between
+a dependency and this set of ``bytes`` objects. They are implemented using
+:func:`functools.singledispatch` and while the types provided by ``web-poet``
+are supported out of the box, user-defined types need a pair of implementation
+functions that need to be registered using
+:func:`web_poet.serialization.register_serialization`.
+
+Fixtures
+========
+
+The provided ``pytest`` plugin expects fixtures in a certain layout. A set of
+fixtures for a single Page Object should be contained in a directory named as
+that Page Object fully qualified class name. Each fixture is a directory inside
+it, that contains data for Page Object inputs and output::
+
+    fixtures
+    └── my_project.po.MyItemPage
+        ├── test-1
+        │   ├── inputs
+        │   │   ├── HttpResponse-body.html
+        │   │   ├── HttpResponse-other.json
+        │   │   └── ResponseUrl.txt
+        │   ├── meta.json
+        │   └── output.json
+        └─── test-2
+            ├── inputs
+            │   ├── HttpResponse-body.html
+            │   ├── HttpResponse-other.json
+            │   └── ResponseUrl.txt
+            ├── meta.json
+            └── output.json
+
+:func:`web_poet.testing.Fixture.save` can be used to create a fixture inside a
+Page Object directory from an iterable of dependencies, an output item and an
+optional metadata dictionary. It can optionally take a name for the fixture
+directory. By default it uses incrementing names "test-1", "test-2" etc.
+
+.. note::
+    ``output.json`` contains a result of
+    ``ItemAdapter(page_object.to_item()).asdict()`` saved as JSON.
+
+After generating a fixture you can edit ``output.json`` to modify expected
+field values and add new fields, which is useful when creating tests for code
+that isn't written yet or before modifying its behavior.
+
+Running tests
+=============
+
+The provided ``pytest`` plugin is automatically registered when ``web-poet`` is
+installed, and running ``pytest`` in a directory containing fixtures will
+discover them and run tests for them.
+
+.. _web-poet-testing-frozen_time:
+
+Handling time fields
+====================
+
+Sometimes output of a page object might depend on the current time. For
+example, the item may contain the scraping datetime, or a current timestamp may
+be used to build some URLs. When a test runs at a different time it will break.
+To avoid this the metadata dictionary can contain a ``frozen_time`` field set
+to the time value used when generating the test. This will instruct the test
+runner to use the same time value so that field comparisons are still correct.
+
+The value can be any string understood by `dateutil`_. If it doesn't include
+timezone information, UTC will be assumed.
```

Comment:
I think we might need to add a bit more information here. In particular:

* Which time format should users use, and which options should they avoid
* What should Windows users do